### PR TITLE
Retry syscalls on EINTR in internal.Connect (fixes: #141)

### PR DESCRIPTION
**Pull request by MaxMaeder**
Retry connect and select syscalls if we get `EINTR` in Internal.Connect().

### CODE REVIEW

File: internal/socket_unix.go [hunk 1]
```diff
@@ -130,40 +130,67 @@ func connect(fd int, remoteAddr net.Addr, timeout time.Duration, opts ...sonicop
...
-		// https://man7.org/linux/man-pages/man2/connect.2.html#EINPROGRESS
+	for {
+		// Try to connect, if it succeeds we're done!
+		err := syscall.Connect(fd, ToSockaddr(remoteAddr))
+		if err == nil {
+			return nil
+		}
+
+		// Retry the connect syscall if interrupted
+		if err == syscall.EINTR {
```
Comment (sergiu128) [inline internal/socket_unix.go:141]: Another improvement is to test for `Errno`s correctly, according to the spec https://pkg.go.dev/syscall#Errno. This means using `errors.Is` pretty much everywhere in this function.

**Review by sergiu128 — CHANGES_REQUESTED** [review]
Looks good, thank you very much! Just needs to handle some edge cases.. and my mistake of not correctly testing for `syscall.Errno`s :)

**Comment by MaxMaeder** [comment]
Should be fixed! I now test for errno correctly.

For the timeouts, I added a specific case in connect() loop so we can return a `sonicerrors.ErrorTimeout`:
```go
// Prevent an infinite connect() loop, time out eventually
if time.Since(startTime) > 5 * timeout {
	return sonicerrors.ErrTimeout
}
```
For the select() loop, I just updated the timeout we pass to the syscall:
```go
t := unix.NsecToTimeval(5 * timeout.Nanoseconds())
```
Linux will update `t` with the time *not slept*, so we don't need an separate check like the other loop.

#46 seems like a similar issue I could tackle, if you had any specific tips.

**Review by sergiu128 — APPROVED** [review]
Excellent solution! Thanks a lot, changed are good! Sorry for the late review, had a lot going on at work the past week.
